### PR TITLE
Add a few useful methods to Pot

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: shared/src/main/scala/crystal/Pot.scala [hunk 1]
```diff
@@ -31,18 +37,16 @@ sealed trait Pot[+A] {
...
+  def toOption: Option[A] = fold(_ => none, _ => none, _.some)
 
   def toTryOption: Option[Try[A]] =
     this match {
       case Pending(_) => none
       case Error(t)   => Failure(t).some
       case Ready(a)   => Success(a).some
     }
+
+  def zoom[B](lens: A => B): Pot[B] = map(lens)
```
Comment (cquiroz) [inline shared/src/main/scala/crystal/Pot.scala:49]: a proper `Lens` cannot be used as we defined `Pot[+A]` and lens is `Lens[A, B]`

File: shared/src/main/scala/crystal/Pot.scala [hunk 2]
```diff
@@ -21,6 +21,12 @@ sealed trait Pot[+A] {
       case Ready(a)       => fr(a)
     }
 
+  def isReady: Boolean = fold(_ => false, _ => false, _ => true)
+
+  def isPending: Boolean = fold(_ =>true, _ => false, _ => true)
```
Comment (rpiaggio) [inline shared/src/main/scala/crystal/Pot.scala:26]: The 3rd parameter should probably be `false` in this case.

File: shared/src/test/scala/crystal/PotSpec.scala [hunk 3]
```diff
@@ -40,20 +40,29 @@ class PotSpec extends DisciplineSuite {
 
   property("Pot[Int].toOption: Pending(_) is None") {
     forAll((l: Long) => Pending(l).toOption.isEmpty)
+  }
 
+  property("Pot[Int].toOption: Pending(_) isPending") {
+    forAll((l: Long) => Pending(l).isPending)
```
Comment (rpiaggio) [inline shared/src/test/scala/crystal/PotSpec.scala:46]: Maybe we can also test for `&& !isReady && !isError`, and similar in the other cases below.

File: shared/src/main/scala/crystal/Pot.scala (hunk 1 above)
Comment (rpiaggio) [reply shared/src/main/scala/crystal/Pot.scala:49]: Is this an alias for `map`? Why do we need it?

Comment (cquiroz) [reply shared/src/main/scala/crystal/Pot.scala:49]: I guess you're right. I was thinking to compose it with a lens but I don't think we can

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**